astropy/coordinates/representation.py

```diff
@@ -1593,6 +1618,47 @@ def represent_as(self, other_class, differential_class=None):
 
         return super().represent_as(other_class, differential_class)
 
+    def transform(self, matrix):
+        r"""Transform the unit-spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) array-like
+            A 3x3 matrix, such as a rotation matrix (or a stack of matrices).
+
+        Returns
+        -------
+        `UnitSphericalRepresentation` or `SphericalRepresentation`
+            If ``matrix`` is O(3) -- :math:`M \dot M^T = I` -- like a rotation,
+            then the result is a `UnitSphericalRepresentation`.
+            All other matrices will change the distance, so the dimensional
+            representation is used instead.
+
+        """
+        # TODO! shortcut if there are differentials.
+        # Currently just super, which uses Cartesian backend.
+        if self.differentials:
```

Comment:
Darn, I just keep on thinking of problems: so, now, if there are differentials but the transformation is not a rotation matrix, we return `UnitSpherical` if there are differentials, but `Spherical` if there are no differentials. This does not seem right... I think we need to have two separate paths:
```
if is_rot:
    if differentials:
        via cartesian
    else:
        direct

else:
    dimensional = ...
    return dimensional.transform(...)
```
I fear this is quite close to what you had before........

Reply:
🤣

Reply:
I reverted it, but just directly attach the differentials ass you suggest below.